agent/src/beerocks/slave/backhaul_manager/wan_monitor.cpp

```diff
@@ -134,7 +134,8 @@ wan_monitor::ELinkState wan_monitor::process()
             struct ifinfomsg *ifi = (struct ifinfomsg *)NLMSG_DATA(hnl);
 
             // Convert the interface index to name
-            if_indextoname(ifi->ifi_index, name);
+            auto iface_index_name = network_utils::linux_get_iface_name(ifi->ifi_index);
+            std::copy_n(iface_index_name.begin(), IFNAMSIZ, name);
```

Comment:
This can access unallocated memory when the interface name is smaller than IFNAMSIZ:
```suggestion
            std::copy_n(iface_index_name.begin(), iface_index_name.length(), name);
```

Relevant for all other commits as well.

Reply:
Fixed